x/liquidity/keeper/migration.go

```
@@ -1,22 +1,169 @@
+		}
+	}()
+
+	cachedCtx, writeCache := ctx.CacheContext()
+	err = ForceWithdrawal(cachedCtx, keeper, bankKeeper, accountKeeper)
+	if err == nil {
+		writeCache()
+	} else {
+		logger.Debug("error occurred on force withdrawal", "error", err)
+	}
+	return
+}
+
+// ForceWithdrawal Forcefully withdraw pool token holders once migration
+func ForceWithdrawal(ctx sdk.Context, keeper Keeper, bankKeeper liquiditytypes.BankKeeper, accountKeeper liquiditytypes.AccountKeeper) error {
+	logger := keeper.Logger(ctx)
+	poolByPoolCoinDenom := map[string]liquiditytypes.Pool{} // PoolCoinDenom => Pool
+	for _, pool := range keeper.GetAllPools(ctx) {
+		poolByPoolCoinDenom[pool.PoolCoinDenom] = pool
+	}
+
+	// Iterate over all the balances of all accounts and find the accounts that hold poolxxx... coin.
+	// Unless it is pool reserve account, forcefully withdraw their pool coin and transfer
+	// the corresponding amount of respective reserve coins back to their accounts.
+	// Lastly, burn pool coins that are withdrawn.
+	accMap := map[string]authtypes.AccountI{}
+	accList := []string{}
+	bankKeeper.IterateAllBalances(ctx, func(address sdk.AccAddress, coin sdk.Coin) (stop bool) {
+		if strings.HasPrefix(coin.Denom, "pool") {
+			pool := poolByPoolCoinDenom[coin.Denom]
+			if _, ok := accMap[address.String()]; !ok {
+				accMap[address.String()] = accountKeeper.GetAccount(ctx, address)
+				accList = append(accList, address.String())
+			}
+
+			// Skip pool reserve accounts
+			acc := accMap[address.String()]
+			if acc.GetSequence() != 0 || acc.GetPubKey() != nil {
+				if _, err := keeper.WithdrawWithinBatch(ctx, &liquiditytypes.MsgWithdrawWithinBatch{
+					WithdrawerAddress: address.String(),
+					PoolId:            pool.GetId(),
+					PoolCoin:          coin,
+				}); err != nil {
+					logger.Debug(
+						"failed force withdrawal",
+						"withdrawer", address.String(),
+						"poolcoin", coin,
+						"error", err,
+					)
```

The error is not being propagated properly. I think it'd be good to store the error in the variable outside of this callback and save the error in that variable. Then after `IterateAllBalances`, we can check if the outside error variable is non-nil.